packages/components/src/components/modal/modal.hbs

```diff
@@ -8,36 +8,43 @@
 <button class="{{@root.prefix}}--btn {{classPrimaryButton}}" type="button" data-modal-target="#modal-{{idSuffix}}">Show
   modal</button>
 
-<div data-modal id="modal-{{idSuffix}}" class="{{@root.prefix}}--modal {{classModalSupplemental}}" role="dialog"
-  aria-modal="true" aria-labelledby="modal-{{idSuffix}}-label" aria-describedby="modal-{{idSuffix}}-heading" tabindex="-1">
-  <div class="{{@root.prefix}}--modal-container">
-    <div class="{{@root.prefix}}--modal-header">
-      <p class="{{@root.prefix}}--modal-header__label {{@root.prefix}}--type-delta" id="modal-{{idSuffix}}-label">Optional label</p>
-      <p class="{{@root.prefix}}--modal-header__heading {{@root.prefix}}--type-beta" id="modal-{{idSuffix}}-heading">Modal heading</p>
-      <button class="{{@root.prefix}}--modal-close" type="button" data-modal-close aria-label="close modal" {{#unless hasFooter}} data-modal-primary-focus{{/unless}}>
-        {{ carbon-icon 'Close16' class=(add @root.prefix '--modal-close__icon') }}
-      </button>
-    </div>
+<div>
```

Comment:
Why did you put additional `<div>` rather than simply adding focus trap element to the last? Note that no user-visible focusable element should be put outside `<div class="bx--modal-container">`.